### PR TITLE
Enable custodians for forward shards

### PR DESCRIPTION
**Pull request by KGibb8**
Just sketching out basic functionality right now. I feel like there's a big code cross-over with what's already been written in the others-shards section and a lot of refactoring can be done with the views to enable these two areas to share code. I'd like to take this on properly when I get back.

Also feeling like we're doing way too much work on fetching and preparing the data in the views. Wondering if a significant portion of this can't be refactored into the back-end and having a single global application state that updates and injecting that into child components / views, rather than doing localised updates in separate views. It would make the front-end significantly easier to extend in the future and extract most of the heavy lifting into `scuttle-dark-crystal`. Again something I'd be super keen to do when I get back. @mixmix perhaps you've already considered this when thinking about your getter and combiner?

I'll add a picture when I've got it working properly.

### CODE REVIEW

**Comment by ameba23** [comment]
i didn't dare click 'yes' even though im pretty sure it doesn't actually publish any messages yet...

File: views/component/are-you-sure.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+module.exports = function AreYouSure (opts) {
+  const {
+    message = '',
+    onSubmit = noOp,
+    onCancel = noOp
+  } = opts
+
+  return h('div.warning', [
+    message ? h('p', message) : '',
+    h('span', 'Are you sure?'),
```
Comment (ameba23) [inline views/component/are-you-sure.js]: Do we wanna ask this before or after they have chosen who to forward them to?

Comment (ameba23) [reply views/component/are-you-sure.js]: or before and after!?

Comment (KGibb8) [reply views/component/are-you-sure.js]: WIP! :)

**Comment by KGibb8** [comment]
This _should_ now work. I've been unable to get it to work as `scuttle-dark-crystal` development and `patchbay-dark-crystal` are not in sync in terms of deliving the API across properly, currently getting `scuttle.recover.pull.replies is not a function`. Any changes that might need to be made should be relatively minimal at this stage to get that working (if any) to this form.

I've done little styling to the `forward/index.js` page at the moment, just getting a form working with some decent styling and warning messages was a higher priority.

The form allows for a user to choose from a friend using Patchbay's `suggest` API. At the moment I haven't set it up to take just a typed `@ABC123...`. It will display the person you're forwarding from, and the person you're forwarding to, and ask you twice to confirm with a different error message (we could probably add an extra `AreYouSure` for luck).

**Comment by KGibb8** [comment]
forward index
forward new

**Comment by KGibb8** [comment]
dark-crystal-forward new-unknown-identity

Unknown identity appears as beige box :smile_cat:

File: views/forward/new.js [hunk 2]
```diff
@@ -39,7 +39,7 @@ module.exports = function DarkCrystalForwardNew (opts) {
     h('section.description', [
       h('div.from', [
         h('span', 'Shards belonging to...'),
-        Recipient({ recp: { name: resolve(name(feedId)), link: feedId }, avatar })
+        Recipient({ recp: { name: (resolve(name(feedId)) || 'unknown'), link: feedId }, avatar })
```
Comment (ameba23) [inline views/forward/new.js]: just to clarify, this is for the case where someone has manually entered a feedId and we cannot find any reference to that feedId (so possible that they entered it wrong).  Just i dont get this `function identity (id) { return id }`.  what does it do?

Cool that you made this functionality though!

**Comment by ameba23** [comment]
This is looking great, good work!  hopefully we can iron out the back end problems soon so we can try this out with actually publishing messages...

only things i can think of to say about it, none of which are major issues:
- +1 for being able to manually enter a feed id.
- one idea would be to only allow forwarding to someone you are following, to make an extra step in the process and reduce the likelihood of error at this stage.  i've no idea how difficult that would be to do with 'suggest'.
- i think the button should say 'forward all shards' and not 'forward' to make things clear.
- a feature that i think would be really cool is if on the modal, after you have chosen the new identity, you see it's age underneath the avatar.  as in the timestamp of the oldest message from that feedid.  Since typically we will be expecting to forward shards to a very recently created account, this could help prevent choosing the wrong feedID.
- if we are gonna have it as a separate page to 'others shards' (and i know this is still under debate) would be good to also be able to see how many shards you are holding for each person with the the little crystal icons. 
- i know this is still kind of a WIP but the linter is barking...

File: views/component/recipient-input.js [hunk 3]
```diff
@@ -0,0 +1,81 @@
+const { h } = require('mutant')
+const addSuggest = require('suggest-box')
+const { isFeedId } = require('ssb-ref')
+
+const MIN_RECPS = 0
+
```
Comment (mixmix) [inline views/component/recipient-input.js]: not sure why this was extracted?

File: views/component/recipient-input.js [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+module.exports = function RecipientInput (opts) {
+  const {
+    state,
+    suggest,
+    maxRecps,
```
Comment (mixmix) [inline views/component/recipient-input.js]: this needs to be capped at 7

File: views/forward/new.js (hunk 2 above)
Comment (mixmix) [reply views/forward/new.js]: this resolve probably needs to change. Need to look closer

**Comment by mixmix** [comment]
This is wired up and working (I fwdd some shards to peg .. .woop!)

There's a bunch of interface stuff still to do but the aim of the game today was to get the functionality live so we can build other things. We can loop back and do a bit more design and some live updating and refactoring of views later

**Pull request merged**